Engine/Fuego/Application.h

```diff
@@ -11,10 +11,11 @@ namespace Fuego
     class Application
     {
     public:
-       FUEGO_API Application();
-       FUEGO_API virtual ~Application();
+		FUEGO_API Application();
+        FUEGO_API virtual ~Application() = default;
+        FUEGO_NON_COPYABLE_NON_MOVABLE(Application)
 
```

Comment:
Why do we need FUEGO_NON_COPYABLE_NON_MOVABLE? Does it relate to .dll export\import issue ?

Reply:
No, it is just the way to comply with the [Rule of Five](https://en.cppreference.com/w/cpp/language/rule_of_three).

Reply:
I used clang static analyser to find all those issues